### PR TITLE
Wire trusted-ca-bundle via /etc/pki

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/openshift/cluster-authentication-operator/pull/172 to make it more standard.

### CODE REVIEW

File: manifests/07_deployment.yaml [hunk 1]
```diff
@@ -23,10 +23,14 @@ spec:
       - name: operator
         image: quay.io/openshift/origin-cluster-authentication-operator:v4.0
         imagePullPolicy: IfNotPresent
-        command: ["authentication-operator", "operator"]
+        command: ["/bin/bash", "ec"]
```
Comment (mfojtik) [inline manifests/07_deployment.yaml]: -ec ?

File: manifests/07_deployment.yaml [hunk 2]
```diff
@@ -23,10 +23,14 @@ spec:
...
         args:
-        - "--config=/var/run/configmaps/config/operator-config.yaml"
-        - "-v=2"
+        - |
+          if [ -s /var/run/configmaps/trusted-ca-bundle/tls-ca-bundle.pem ]; then
```
Comment (mfojtik) [inline manifests/07_deployment.yaml]: I wish the operator does this during startup and we don't have to hack ways in bash.

Comment (sttts) [reply manifests/07_deployment.yaml]: it cannot. Then it is too late. And this does not belong into a binary. It is system plumbing. It is just right here.

Comment (mfojtik) [reply manifests/07_deployment.yaml]: if `tls-ca-bundle.pem` is empty, will that brick the operator?

Comment (sttts) [reply manifests/07_deployment.yaml]: it is non-existent if not part of the configmap.

Comment (sttts) [reply manifests/07_deployment.yaml]: The network operator validates it.

**Comment by enj** [comment]
After fixing the comment thing and making sure the bump size is expected, feel free to have anyone tag.

File: pkg/operator2/operator.go [hunk 3]
```diff
@@ -113,7 +114,7 @@ const (
 	// trustedCABundleName part of manifests, if changing this, need to change that, too
 	trustedCABundleName      = systemConfigPrefix + "trusted-ca-bundle"
 	trustedCABundleKey       = "ca-bundle.crt"
-	trustedCABundleMountDir  = "/etc/pki/ca-trust/extracted/pem"
+	trustedCABundleMountDir  = systemTrustStoreDirPath
```
[the remark this answers is not in the document]
Comment (sttts) [reply pkg/operator2/operator.go]: this is a revert. Would avoid to change anything not neccesary.

**Comment by sttts** [comment]
> nit. Why is the glide bump so large?

because glide.

File: manifests/07_deployment.yaml (hunk 2 above)
Comment (sttts) [reply manifests/07_deployment.yaml]: @stlaz noticed that we must not copy empty files if I have understood him correctly.

File: pkg/operator2/operator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sttts) [reply pkg/operator2/operator.go]: I reverted the changes in that file, yes.

**Comment by enj** [comment]
> I reverted the changes in that file, yes.

Not sure I understand but do not feel strongly to block anything.

/lgtm

File: pkg/operator2/operator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sttts) [reply pkg/operator2/operator.go]: Have removed the constant now.

**Pull request merged**